native/c/zcli.hpp

```diff
@@ -453,55 +446,55 @@ void ZCLI::help()
   }
 }
 
-ZCLIGroup &ZCLI::get_group(string group_name)
+ZCLIGroup *ZCLI::get_group(string group_name)
 {
   for (vector<ZCLIGroup>::iterator it = groups.begin(); it != groups.end(); it++)
   {
     if (group_name == it->get_name())
-      return *it;
+      return &*it;
     for (vector<string>::iterator iit = it->get_aliases().begin(); iit != it->get_aliases().end(); iit++)
     {
       if (group_name == *iit)
-        return *it;
+        return &*it;
     }
   }
-  return ZCLI::group_not_found;
+  return nullptr;
 }
 
-ZCLIVerb &ZCLIGroup::get_verb(string verb_name)
+ZCLIVerb *ZCLIGroup::get_verb(string verb_name)
 {
   for (vector<ZCLIVerb>::iterator it = verbs.begin(); it != verbs.end(); it++)
   {
     if (verb_name == it->get_name())
-      return *it;
+      return &*it;
     for (vector<string>::iterator iit = it->get_aliases().begin(); iit != it->get_aliases().end(); iit++)
     {
       if (verb_name == *iit)
-        return *it;
+        return &*it;
     }
   }
-  return ZCLIGroup::verb_not_found;
+  return nullptr;
 }
 
-ZCLIOption &ZCLIOptionProvider::get_option(string option_name)
+ZCLIOption *ZCLIOptionProvider::get_option(string option_name)
 {
   for (vector<ZCLIOption>::iterator it = options.begin(); it != options.end(); it++)
   {
     vector<string> &aliases = it->get_aliases();
     if (option_name == it->get_flag_name() || std::find(aliases.begin(), aliases.end(), option_name) != it->get_aliases().end())
-      return *it;
+      return &*it;
   }
-  return ZCLIOptionProvider::option_not_found;
+  return nullptr;
 }
 
-ZCLIPositional &ZCLIPositionalProvider::get_positional(string positional_name)
+ZCLIPositional *ZCLIPositionalProvider::get_positional(string positional_name)
 {
   for (vector<ZCLIPositional>::iterator it = positionals.begin(); it != positionals.end(); it++)
   {
     if (positional_name == it->get_name())
-      return *it;
+      return &*it;
   }
-  return ZCLIPositionalProvider::positional_not_found;
+  return nullptr;
```

Comment:
thanks for returning the `nullptr` instead.
I believe this is part of the refactoring effort that you and Trae were discussing offline. 🙏